### ISSUE TITLE
Some challenges when upgrading to Ruby 2.5.1

### ISSUE THREAD

**Issue by mottalrd** [auth0/ruby-auth0#122]
Hello and thank you for this library, super helpful.

We are trying to upgrade to Ruby 2.5.1 but we are experiencing failures on Circle CI.
What Ruby version are you running on Travis for this gem?

What would be the best way for us to help you with that?  My plan would be:
* Create a new ruby (maybe rails) app
* Create a new open Auth0 env for such testing (do you have an open one already?)
* Setup Circle CI 
* Add a test that fails with 2.5.1

but that requires a bit of investment on our side so before moving forward I would love to get some feedback from you on that.

Thank you, much appreciated

**Comment by mottalrd** [comment]
Just to add a bit more information, the issue seems to belong to rest-client (a dependency of this gem) and this is the error we get

`KeyError: key not found: :ciphers`

**Comment by mottalrd** [comment]
Hello, any feedback on this? @joshcanhelp ?

**Comment by joshcanhelp** [comment]
@mottalrd - Sorry for the delay here, I was not set to watch this repo for some reason. 

Tested versions for CI are listed here in the repo:

https://github.com/auth0/ruby-auth0/blob/master/.travis.yml#L5

> Create a new open Auth0 env for such testing (do you have an open one already?)

I'm not totally clear what you're asking. Can you rephrase please?

Just to be clear, is your main question "does this work with Ruby 2.5.1?" If so, I can say that I'm running Ruby `2.5.1` while developing and testing. 

It looks like the `rest-client` gem has a 2.1.0 RC. We'll keep an eye on that and test + upgrade when it's out. 

Thanks!

**Comment by joshcanhelp** [comment]
@mottalrd - No problem. I'll add that and test first thing next week 👍

**Comment by joshcanhelp** [comment]
@mottalrd - Apologies for the delay. Everything seems to have gone through as expected on our end:

https://travis-ci.org/auth0/ruby-auth0/builds/413753024

The test suite on this SDK takes quite a while so we'll need to address that before we can add more Ruby versions but I'll leave that PR open for you to take a look.

**Comment by mottalrd** [comment]
Thank you @joshcanhelp , this probably means that our failure is related to CircleCI, this helps quite a lot to narrow it down.

**Comment by joshcanhelp** [comment]
@mottalrd - No problem! We'll add 2.5.1 at some point soon. Feel free to post back here with any other questions or close this out if there's nothing needed on our end.
